File: 0739-daily-temperatures/0739-daily-temperatures.py
class Solution:
    def dailyTemperatures(self, temperatures: List[int]) -> List[int]:
        curr_indx = len(temperatures) - 1
        stk = []
        res = [0] * len(temperatures)
        while curr_indx >= 0:
            if stk:
                while stk and stk[-1][0] <= temperatures[curr_indx]:
                    stk.pop()
                if stk:
                    res[curr_indx] = stk[-1][-1] - curr_indx
                else:
                    res[curr_indx] = 0
            stk.append([temperatures[curr_indx],curr_indx])
            curr_indx -= 1
        return res
                

File: 0739-daily-temperatures/0739-daily-temperatures.java
class Solution {
    public int[] dailyTemperatures(int[] temperatures) {
        int [] warmerDays = new int[temperatures.length];
        Stack<Integer> warmerTemp = new Stack<>();
        for(int i = temperatures.length - 1; i >= 0 ; i -= 1)
        {
            if(warmerTemp.size() == 0)
            {
                warmerTemp.push(i);
                warmerDays[i] = 0;
            }
            else
            {
                while ((!warmerTemp.isEmpty()) && (temperatures[warmerTemp.peek()] <= temperatures[i]))
                {
                    warmerTemp.pop();

                }
                if (warmerTemp.isEmpty())
                {
                    warmerDays[i] = 0;
                }
                else
                {
                    warmerDays[i] = warmerTemp.peek() - i;
                }
                warmerTemp.push(i);
            }

        }
        return warmerDays;
    }
}